packages/dd-trace/src/appsec/rasp/ssrf.js

```diff
@@ -25,16 +41,108 @@ function analyzeSsrf (ctx) {
 
   if (!req || !outgoingUrl) return
 
+  // Determine if we should collect the response body based on sampling rate
+  const includeBodies = downstream.shouldSampleBody(req)
+
+  // Initialize state for tracking this request's response
+  ctx[RESPONSE_STATE] = {
+    req,
+    includeBodies,
+    chunks: includeBodies ? [] : null,
+    done: false
+  }
+
+  const requestAddresses = downstream.extractRequestData(ctx)
+
   const ephemeral = {
-    [addresses.HTTP_OUTGOING_URL]: outgoingUrl
+    [addresses.HTTP_OUTGOING_URL]: outgoingUrl,
+    ...requestAddresses
   }
 
-  const raspRule = { type: RULE_TYPES.SSRF }
+  const raspRule = { type: RULE_TYPES.SSRF, variant: 'request' }
 
   const result = waf.run({ ephemeral }, req, raspRule)
 
-  const res = store?.res
-  handleResult(result, req, res, ctx.abortController, config, raspRule)
+  handleResult(result, req, store?.res, ctx.abortController, config, raspRule)
+
+  downstream.incrementDownstreamAnalysisCount(req)
+
+  // Track body analysis count if we're sampling the response body
+  if (includeBodies) {
+    downstream.incrementBodyAnalysisCount(req)
+  }
+}
+
+/**
+ * Collects outgoing response chunks when body sampling is enabled.
+ * @param {{
+ * ctx: object,
+ * chunk: Buffer|string|Uint8Array,
+ * res: import('http').IncomingMessage}} payload event payload from the channel.
+ */
+function handleResponseData ({ ctx, chunk, res }) {
```

Comment:
Not sure if we should be doing this here, or in `client.js`. This whole logic is `http` module dependant, we should try to do module dependant logic in the module instrumentation, this could be different for other lib/frameworks.